File: src/Reservations.Service/Models/Reservation.cs
using System.ComponentModel;
using Reservations.Service.Attributes;

namespace Reservations.Service.Models
{
    public static class ReservationModelKeys
    {
        public const string SlotId = "slotId";
        public const string ReservedById = "reservedById";
    }

    public class Reservation
    {
        public Guid Id { get; set; }

        [Filterable(ReservationModelKeys.SlotId)]
        public Guid SlotId { get; set; }

        [Filterable(ReservationModelKeys.ReservedById)]
        public Guid ReservedById { get; set; }

        public ReservationPeriod Period { get; set; }

        public ReservationStatus Status { get; set; }
    }

  
}

File: src/Web/web/src/models/reservation.ts
export enum ReservationStatus {
  pending = 0,
  confirmed = 1,
  rejected = 2,
}

//TODO extract reservation for slot and for reservation
export interface Reservation {
  id: string;
  reservedById: string;
  slotId: string;
  status: ReservationStatus;
  period: { from: string; to: string };
}
